pegged/peg.d

```diff
@@ -1755,8 +1758,12 @@ template or(rules...) if (rules.length > 0)
     }
 }
 
-auto getUpto(ParseTree[] children, size_t minFailedLength) pure nothrow {
-    return children.filter!(r => max(r.end, r.failEnd) >= minFailedLength).array();
+auto getUpto(scope ParseTree[] children, size_t minFailedLength) pure nothrow {
+    auto ret = children.filter!(r => max(r.end, r.failEnd) >= minFailedLength).array();
+
+    // FIXME: .array() garantees a copy or the elements. Phobos/compiler scope
+    // inference issue.
+    return (() @trusted => ret)();
```

Comment:
The problem here, I am guessing, is that `array` does not produce a deep copy. `ParseTree.input` for example still refers to the original input, and it should be. Again I think you are looking for `return`, not `scope`.

Reply:
So making this `@trusted` would be a lie, is my thought.

Reply:
> The problem here, I am guessing, is that `array` does not produce a deep copy. `ParseTree.input` for example still refers to the original input, and it should be. Again I think you are looking for `return`, not `scope`.

Nope. If you try with `.dup` it works, so you can prove that. I just didn't do .array.dup, as .array already allocates a new array. See @dkorpel .